lib/slots.ts

```diff
@@ -275,9 +275,10 @@ export function HasSlot(container: unknown, ...ids: (keyof Slots)[]): boolean {
   return !!myslots && ids.reduce((all: boolean, id) => all && id in myslots, true);
 }
 export function GetSlot<KeyT extends keyof Slots>(
-  container: Slots[typeof id]['usedBy'],
+  container: Slots[typeof id]['usedBy'] | undefined,
```

Comment:
I think it signifies a bug in the polyfill if we end up calling `GetSlot(undefined, ...)` so I'd be happy if TS would complain at compile time rather than resorting to a runtime error. I wonder if there's any way to do that?

Reply:
In this case it can be `undefined` if an incorrect disambiguation is given to `BuiltinTimeZoneGetInstantFor`, so maybe it would be best to just throw?

Reply:
Maybe we can tighten the types in BuiltinTimeZoneGetInstantFor to ensure at compile time that no incorrect disambiguation can be given there? Or otherwise throw earlier if that happens?

Reply:
Seems like there may be a legit issue with `DisambiguatePossibleInstants` because the spec says there should be an assertion that the `disambiguation` value is valid but the polyfill lacks this assertion. We could add a line at the end of that method e.g. 
```js
    throw new Error(`assertion failed: invalid disambiguation value ${disambiguation}`);
```